python/src/energuide/snippets.py

```diff
@@ -95,12 +95,54 @@ def _wall_code_snippet(wall_code: etree.ElementTree) -> typing.Dict[str, typing.
         'componentTypeSizeFrench': component_type_size_french,
     }
 
+def _window_code_snipped(window_code: etree.ElementTree) -> typing.Dict[str, typing.Any]:
+    code_id = window_code.attrib['id']
+    label = window_code.findtext('Label')
+
+    glazing_types_english = window_code.findtext('Layers/GlazingTypes/English')
+    glazing_types_french = window_code.findtext('Layers/GlazingTypes/French')
+
+    coatings_tints_english = window_code.findtext('Layers/CoatingsTints/English')
+    coatings_tints_french = window_code.findtext('Layers/CoatingsTints/French')
+
+    fill_type_english = window_code.findtext('Layers/FillType/English')
+    fill_type_french = window_code.findtext('Layers/FillType/French')
+
+    spacer_type_english = window_code.findtext('Layers/SpacerType/English')
+    spacer_type_french = window_code.findtext('Layers/SpacerType/French')
+
+    type_english = window_code.findtext('Layers/Type/English')
+    type_french = window_code.findtext('Layers/Type/French')
+
+    frame_material_english = window_code.findtext('Layers/FrameMaterial/English')
+    frame_material_french = window_code.findtext('Layers/FrameMaterial/French')
+
+    return {
+        'id': code_id,
+        'label': label,
+        'glazingTypeEnglish': glazing_types_english,
+        'glazingTypeFrench': glazing_types_french,
+        'coatingsTintsEnglish': coatings_tints_english,
+        'coatingsTintsFrench': coatings_tints_french,
+        'fillTypeEnglish': fill_type_english,
+        'fillTypeFrench': fill_type_french,
+        'spacerTypeEnglish': spacer_type_english,
+        'spacerTypeFrench': spacer_type_french,
+        'typeEnglish': type_english,
+        'typeFrench': type_french,
+        'frameMaterialEnglish': frame_material_english,
+        'frameMaterialFrench': frame_material_french,
```

Comment:
Are all of these required in the HOIS? Let's make sure to start from exactly what's necessary to replicate the information contained there. We can add to it later.

Reply:
They're all required, the HOIS has a `TYPE` column that is a combination of all six of these fields

Reply:
👍